### PR TITLE
Updating UI for create acceleration flyout

### PR DESCRIPTION
**Pull request by ps48**
### Description
NOTE: Major file changes are for a directory change
* Updating UI for create acceleration flyout  - General UI updates and changes from backend API

### Issues Resolved
1. Add bloom filter for skipping index type
2. Add window.start column as option in materialized views
3. Add support for incremental refresh
4. Add support for preselected datasources for the accelerations flyout
5. Add advanced settings accordian and index settings 
6. Update jest tests

### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

**Review by mengweieric — APPROVED** [review]

File: public/components/datasources/components/manage/accelerations/create_accelerations_flyout/selectors/source_selector.tsx [hunk 1]
```diff
@@ -0,0 +1,270 @@
...
+              <EuiFlexItem>
+                <EuiComboBox
+                  placeholder="Select a database"
+                  singleSelection={{ asPlainText: true }}
+                  options={databases}
+                  selectedOptions={selectedDatabase}
+                  onChange={(databaseOptions) => {
+                    if (databaseOptions.length > 0) {
+                      setAccelerationFormData(
+                        producer((accData) => {
```
Comment (paulstn) [inline public/components/datasources/components/manage/accelerations/create_accelerations_flyout/selectors/source_selector.tsx:203]: Is producer here being used to create a new object from acceleration data? If so, why is it needed as opposed to mutating accData?

File: public/framework/catalog_cache/cache_loader.tsx [hunk 2]
```diff
@@ -123,11 +129,53 @@ export const updateAccelerationsToCache = (
...
+export const updateTableColumnsToCache = (
+  dataSourceName: string,
+  databaseName: string,
+  tableName: string,
+  pollingResult: AsyncPollingResult
+) => {
+  const cachedDatabase = CatalogCacheManager.getDatabase(dataSourceName, databaseName);
+  const currentTime = new Date().toUTCString();
+
+  if (!pollingResult) {
```
Comment (paulstn) [inline public/framework/catalog_cache/cache_loader.tsx:141]: Why not move this to the first line of the method? I would assume `CatalogCacheManager.getDatabase` is expensive, and `pollingResult` doesn't rely on that, so wouldn't it be better to quickly exit if there's no result yet?

Comment (ps48) [reply public/framework/catalog_cache/cache_loader.tsx:138]: good catch! updated this.

File: public/components/datasources/components/manage/accelerations/create_accelerations_flyout/selectors/source_selector.tsx (hunk 1 above)
Comment (ps48) [reply public/components/datasources/components/manage/accelerations/create_accelerations_flyout/selectors/source_selector.tsx:203]: accData/AccelerationFormData contains large number of objects as it contains whole forms data. Hence, we rely on `immer` library provided `producer` function to update the state. This helps to do away with complex nesting stuff.

File: public/components/datasources/components/manage/accelerations/create_accelerations_flyout/selectors/index_setting_options.tsx [hunk 3]
```diff
@@ -38,61 +36,74 @@ export const IndexSettingOptions = ({
...
-      id: autoRefreshId,
-      label: 'Auto refresh',
+      value: 'auto',
+      inputDisplay: 'Auto',
+      dropdownDisplay: (
+        <Fragment>
+          <strong>Auto</strong>
+          <EuiText size="s" color="subdued">
+            <p className="EuiTextColor--subdued">
+              Automatically refreshes the index when new data is available.
```
[the remark this answers is not in the document]
Comment (ps48) [reply public/components/datasources/components/manage/accelerations/create_accelerations_flyout/selectors/index_setting_options.tsx:48]: Created an issue to follow up: https://github.com/opensearch-project/dashboards-observability/issues/1539 
Ideally we should have a meta issue for all of observability to do this.

File: public/components/datasources/components/manage/accelerations/create_accelerations_flyout/create/create_acceleration.tsx [hunk 4]
```diff
@@ -83,8 +96,69 @@ export const CreateAcceleration = ({
...
     },
   });
+  const [tableFieldsLoading, setTableFieldsLoading] = useState(false);
+  const { loadStatus, startLoading } = useLoadTableColumnsToCache();
+
+  const loadColumnsToAccelerationForm = (cachedTable: CachedTable) => {
+    const idPrefix = htmlIdGenerator()();
+    const dataTableFields = cachedTable.columns!.map((col, index: number) => ({
+      ...col,
+      id: `${idPrefix}${index + 1}`,
```
[the remark this answers is not in the document]
Comment (ps48) [reply public/components/datasources/components/manage/accelerations/create_accelerations_flyout/create/create_acceleration.tsx:106]: running html generator is loop is costly. Hence, kept the prefix out.

File: public/components/datasources/components/manage/accelerations/create_accelerations_flyout/visual_editors/materialized_view/group_by_tumble_expression.tsx [hunk 5]
```diff
@@ -36,7 +39,7 @@ export const GroupByTumbleExpression = ({
   const [groupbyValues, setGroupByValues] = useState<GroupByTumbleType>({
     timeField: '',
     tumbleWindow: 1,
-    tumbleInterval: ACCELERATION_TIME_INTERVAL[0].value,
+    tumbleInterval: ACCELERATION_TIME_INTERVAL[2].value,
```
[the remark this answers is not in the document]
Comment (ps48) [reply public/components/datasources/components/manage/accelerations/create_accelerations_flyout/visual_editors/materialized_view/group_by_tumble_expression.tsx:42]: Added a comment now `// minute` . I do think enum makes sense to be used here. Will update this in future PRs.

File: public/components/datasources/components/manage/accelerations/create_accelerations_flyout/create/create_acceleration.tsx [hunk 6]
```diff
@@ -52,21 +65,21 @@ export const CreateAcceleration = ({
       groupByTumbleValue: {
         timeField: '',
         tumbleWindow: 0,
-        tumbleInterval: '',
+        tumbleInterval: ACCELERATION_TIME_INTERVAL[2].value,
```
[the remark this answers is not in the document]
Comment (ps48) [reply public/components/datasources/components/manage/accelerations/create_accelerations_flyout/create/create_acceleration.tsx]: same as above.

File: public/components/datasources/components/manage/accelerations/create_accelerations_flyout/create/create_acceleration.tsx (hunk 4 above)
[the remark this answers is not in the document]
Comment (paulstn) [reply public/components/datasources/components/manage/accelerations/create_accelerations_flyout/create/create_acceleration.tsx:106]: That makes sense

File: public/components/datasources/components/manage/accelerations/create_accelerations_flyout/selectors/source_selector.tsx (hunk 1 above)
Comment (paulstn) [reply public/components/datasources/components/manage/accelerations/create_accelerations_flyout/selectors/source_selector.tsx:203]: Sounds good

**Review by paulstn — APPROVED** [review]

**Pull request merged**